### ISSUE TITLE
tiff file handle is closed too early

### ISSUE THREAD

**Issue by db4** [python-pillow/Pillow#2049]
Suppose I have a TIFF file with 2 pages. I do

```
tiff = Image.open('file.tif')
tiff.seek(0)
tiff.load()
tiff.seek(1)
tiff.load()
```

this somehow triggers the tiff handle closing:

```
# TiffImagePlugin.py, line 1077
        # libtiff closed the fp in a, we need to close self.fp, if possible
        if hasattr(self.fp, 'close'):
            if not self.__next:
                self.fp.close()
```

and subsequent

```
tiff.seek(0)
```

will fail with `ValueError: seek of closed file`
I tried to fix this in https://github.com/python-pillow/Pillow/pull/2047/commits/30dc18523676c35d7610f69be4fca6c5d3bbb2a5 but this was not enough. So I hope someone familiar with the code will fix it correctly.

**Comment by wiredfool** [comment]
Does this happen with any of the existing multi page tiff images that we're shipping as tests, or is there a different tiff to use as a test case which would expose this?

**Comment by db4** [comment]
My impression is that we need an additional test file. You can find it here together with a test case: https://github.com/python-pillow/Pillow/pull/2047/commits/25804cc8c753eb542f66310c0ef6fefff99d97ab

**Comment by sekrause** [comment]
The file descriptor handling with multipage TIFFs is actually quite messed up, see my realted issue #1651 and especially my latest comment there: https://github.com/python-pillow/Pillow/issues/1651#issuecomment-250454257

So we have both the problems of `self.fd` being closed too early and at the same time dangling duplicated file descriptiors which are not being closed at any time.

The solution should probably look like this:

- Only close `self.fd` when the image file is being closed.
- Only create duplicate file descriptors when we need to pass a file descriptors into libtiff (because libtiff closes the file handles passed to it).

**Comment by radarhere** [comment]
Using https://github.com/python-pillow/Pillow/blob/25804cc8c753eb542f66310c0ef6fefff99d97ab/Tests/images/compression2.tif?raw=true and testing, this looks like it was fixed by #2330
